### PR TITLE
Check equality of proposer slashing signed block header messages

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Fix minor state transition bug: check equality of headers's messages during `process_proposer_slashings`.

## Fixed Issue(s)

https://github.com/sigp/beacon-fuzz/issues/90

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: ethereum/core/src/main/java/tech/pegasys/teku/core/operationvalidators/ProposerSlashingStateTransitionValidator.java [hunk 1]
```diff
@@ -45,7 +45,9 @@
                 ProposerSlashingInvalidReason.PROPOSER_INDICES_DIFFERENT),
         () ->
             check(
-                !Objects.equals(proposerSlashing.getHeader_1(), proposerSlashing.getHeader_2()),
+                !Objects.equals(
+                    proposerSlashing.getHeader_1().getMessage(),
+                    proposerSlashing.getHeader_2().getMessage()),
```
Comment (mbaxter) [inline ethereum/core/src/main/java/tech/pegasys/teku/core/operationvalidators/ProposerSlashingStateTransitionValidator.java]: (nit) We can simplify:
```suggestion
                !Objects.equals(
                    header1.getMessage(),
                    header2.getMessage()),
```

File: networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/ProposerSlashingValidatorTest.java [hunk 2]
```diff
@@ -117,4 +119,26 @@ public void shouldIgnoreProposerSlashingForTheSameProposer() throws Exception {
...
+    beaconChainUtil.initializeStorage();
+    beaconChainUtil.createAndImportBlockAtSlot(6);
+    stateTransitionValidator = new ProposerSlashingStateTransitionValidator();
+    SignedBeaconBlockHeader signedBeaconBlockHeader =
+        dataStructureUtil.randomSignedBeaconBlockHeader();
+    ProposerSlashing slashing =
+        new ProposerSlashing(
+            new SignedBeaconBlockHeader(
+                signedBeaconBlockHeader.getMessage(), BLSSignature.random(100)),
+            signedBeaconBlockHeader);
```
Comment (mbaxter) [inline networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/ProposerSlashingValidatorTest.java]: (nit) I'd declare the second header as a variable, and assert that the 2 headers are not equal, just so we're being really explicit and verifying our assumptions:

```suggestion
    final SignedBeaconBlockHeader header2 = new SignedBeaconBlockHeader(signedBeaconBlockHeader.getMessage(), BLSSignature.random(100);
    assertThat(header2).isNotEqualTo(signedBeaconBlockHeader);
    ProposerSlashing slashing =
        new ProposerSlashing(
            header2,
            signedBeaconBlockHeader);
```

**Review by mbaxter — APPROVED** [review]

File: ethereum/core/src/main/java/tech/pegasys/teku/core/operationvalidators/ProposerSlashingStateTransitionValidator.java (hunk 1 above)
Comment (cemozerr) [reply ethereum/core/src/main/java/tech/pegasys/teku/core/operationvalidators/ProposerSlashingStateTransitionValidator.java]: Done.

File: networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/ProposerSlashingValidatorTest.java (hunk 2 above)
Comment (cemozerr) [reply networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/ProposerSlashingValidatorTest.java]: Done.

**Pull request merged**